reconcilers/flow_test.go

```diff
@@ -269,6 +271,132 @@ func TestWhile(t *testing.T) {
 	})
 }
 
+func TestForEach(t *testing.T) {
+	testNamespace := "test-namespace"
+	testName := "test-resource"
+
+	scheme := runtime.NewScheme()
+	_ = resources.AddToScheme(scheme)
+
+	resource := dies.TestResourceBlank.
+		MetadataDie(func(d *diemetav1.ObjectMetaDie) {
+			d.Namespace(testNamespace)
+			d.Name(testName)
+		}).
+		SpecDie(func(d *dies.TestResourceSpecDie) {
+			d.Fields(map[string]string{})
+		})
+
+	rts := rtesting.SubReconcilerTests[*resources.TestResource]{
+		"empty items": {
+			Resource:       resource.DieReleasePtr(),
+			ExpectResource: resource.DieReleasePtr(),
+		},
+		"each container": {
+			Resource: resource.
+				SpecDie(func(d *dies.TestResourceSpecDie) {
+					d.TemplateDie(func(d *diecorev1.PodTemplateSpecDie) {
+						d.SpecDie(func(d *diecorev1.PodSpecDie) {
+							d.ContainerDie("hello", func(d *diecorev1.ContainerDie) {
+								d.Image("world")
+							})
+							d.ContainerDie("foo", func(d *diecorev1.ContainerDie) {
+								d.Image("bar")
+							})
+						})
+					})
+				}).
+				DieReleasePtr(),
+			ExpectResource: resource.
+				SpecDie(func(d *dies.TestResourceSpecDie) {
+					d.TemplateDie(func(d *diecorev1.PodTemplateSpecDie) {
+						d.SpecDie(func(d *diecorev1.PodSpecDie) {
+							d.ContainerDie("hello", func(d *diecorev1.ContainerDie) {
+								d.Image("world")
+							})
+							d.ContainerDie("foo", func(d *diecorev1.ContainerDie) {
+								d.Image("bar")
+							})
+						})
+					})
+				}).
+				StatusDie(func(d *dies.TestResourceStatusDie) {
+					// container.name -> container.image-index-length
+					d.AddField("hello", "world-0-2")
+					d.AddField("foo", "bar-1-2")
+				}).
+				DieReleasePtr(),
+		},
+		"terminate iteration on error": {
+			Resource: resource.
+				SpecDie(func(d *dies.TestResourceSpecDie) {
+					d.TemplateDie(func(d *diecorev1.PodTemplateSpecDie) {
+						d.SpecDie(func(d *diecorev1.PodSpecDie) {
+							d.ContainerDie("hello", func(d *diecorev1.ContainerDie) {
+								d.Image("world")
+							})
+							d.ContainerDie("die", func(d *diecorev1.ContainerDie) {
+								d.Image("die")
```

Comment:
**[issue, absolutely non-blocking]:** overloading the term "die" with "exit early" is fine, but makes the reader squint

Reply:
the other meaning of "die" :)